apple/internal/transition_support.bzl

```diff
@@ -658,15 +659,22 @@ def _xcframework_transition_impl(settings, attr):
 
     # TODO(b/288582842): Update for visionOS when we're ready to support it in XCFramework rules.
     for platform_type in ["ios", "tvos", "visionos", "watchos", "macos"]:
-        if not hasattr(attr, platform_type):
+        platform_attr = getattr(attr, platform_type, None)
```

Comment:
Use `getattr` with a `None` default instead of `hasattr` which will still return true even when `ios` for example is the default value of `{}`, this was causing a `apple_xcframework` with only a `macos` attr to also go through this transition for the `ios` platform. 

With `getattr` we correctly get the value, for example the empty dict `{}`, which then matches on the `if not` check (it's an empty dict which is falsy), where `hasattr` would've returned true since the `ios` attr is technically available.